dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -22,19 +22,22 @@
 /**
  *
  */
-
 public record ToolMetadata(Tool annotation, Bean<?> bean, AnnotatedMethod<?> method,
-                           Map<String, ArgumentMetadata> arguments, String name, String title, String description) {
+                           Map<String, ArgumentMetadata> arguments,
+                           Map<String, SpecialArgumentMetadata> specialArguments,
+                           String name, String title, String description) {
 
     public record ArgumentMetadata(Type type, int index, String description) {}
 
+    public record SpecialArgumentMetadata(Type type, int index) {}
```

Comment:
There's going to be a limited number of permitted special argument types. If we create an enum for them and use that here instead of `Type`, the compiler can validate for us we've covered all cases when we `switch` on the enum.